explorer/interpreter/interpreter.cpp

```diff
@@ -1087,6 +1087,56 @@ auto Interpreter::CallFunction(const CallExpression& call,
   }
 }
 
+// Returns true if the format string is okay to pass to formatv. This only
+// supports `{{` and `{N}` as special syntax.
+static auto IsValidFormatString(const char* format_string, int num_args)
+    -> bool {
+  const char* cursor = format_string;
+  while (true) {
+    switch (*cursor) {
+      case '\0':
+        // End of string.
+        return true;
+      case '{':
+        // `{` is a special character.
+        ++cursor;
+        switch (*cursor) {
+          case '\0':
+            // Early end.
+            return false;
+          case '{':
+            // Escaped `{`.
+            ++cursor;
+            break;
+          case '}':
+            // Reject `{}`.
+            return false;
+          default:
+            int index = 0;
+            while (*cursor != '}') {
+              if (*cursor < '0' || *cursor > '9') {
+                // Non-numeric text, including `\0`.
+                return false;
+              }
+              index = (10 * index) + (*cursor - '0');
+              ++cursor;
+            }
+            if (index >= num_args) {
+              // Out-of-bounds.
+              return false;
+            }
+            // Move past the `}`.
+            ++cursor;
+        }
+        break;
+      default:
```

Comment:
`Print("}")` passes but shouldn't. Because `}` needs be to escaped as `}}`

Reply:
on reading the docs more carefully, "}" is allowed by `llvm::format_v`. But, it still feels weird.

Reply:
Thanks, this is not actionable.